includes/admin/OrderSettings.php

```diff
@@ -272,19 +272,21 @@ private function setAllData(): void
     }
 
     /**
-     * @TODO MY-28781 refactor OrderSettings-> get and set Shipping and Billing Recipients
-     *
      * @return self
      */
     public function setShippingRecipient(): self
     {
+        $recipient = WCMP_Export::getRecipientFromOrder($this->order);
+
         $this->shippingRecipient = (new Recipient())
-            ->setCc($this->order->get_shipping_country())
-            ->setCity($this->order->get_shipping_city())
-            ->setCompany($this->order->get_shipping_company())
-            ->setPerson(substr($this->order->get_formatted_shipping_full_name(), 0, 50))
-            ->setPostalCode($this->order->get_shipping_postcode())
-            ->setStreet($this->makeStreet($this->order->get_billing_address_1(), $this->order->get_billing_address_2()));
+            ->setCc($recipient['cc'])
+            ->setCity($recipient['city'])
+            ->setCompany($recipient['company'])
+            ->setPerson(substr($recipient['person'], 0, 50))
```

Comment:
Moeten we dit niet ook door de sdk laten doen?

Reply:
Uiteindelijk wel